src/libraries/System.Net.Http/src/Resources/Strings.resx

```diff
@@ -211,6 +211,9 @@
     <value>A request message must be provided. It cannot be null.</value>
   </data>
   <data name="net_http_message_not_success_statuscode" xml:space="preserve">
+    <value>Response status code does not indicate success: {0}.</value>
```

Comment:
Is this message susceptible to improvement? Seems clumsy. We only show this when code is not 2XX - is that useful to spell out?

"HTTP status code {0} ({1})" would be sufficient?

Reply:
> "HTTP status code {0} ({1})" would be sufficient?

For someone who doesn't know that 2xx is success and other codes are non-success, I don't think that conveys as well why you're getting an exception.

Reply:
we _could_ possibly split them by the groups eg 2xx, 3xx, 4xx since they all have distinct general reasons....